### ISSUE TITLE
fails to parse async arrow functions w/ eslint@2.10.0

### ISSUE THREAD

**Issue by jonathanong** [babel/babel-eslint#311]
test case: https://travis-ci.org/jongleberry/eslint-config-jongleberry/builds/130127232 https://github.com/jongleberry/eslint-config-jongleberry/pull/50

`arrow-functions.js`:

```js
export const double = async a => a * 2
export const multiply = async (a, b) => a * b
```

`async.js`:

```js
export default async (a, b) => {
  return await Promise.resolve(a + b)
}
```

error:

```bash
/home/travis/build/jongleberry/eslint-config-jongleberry/test/fixtures/arrow-functions.js
  1:29  error  Parsing error: Unexpected token a
/home/travis/build/jongleberry/eslint-config-jongleberry/test/fixtures/async.js
  1:29  error  Parsing error: Unexpected token =>
✖ 2 problems (2 errors, 0 warnings)
```

**Comment by not-an-aardvark** [comment]
I can confirm this issue. However, it seems to occur with all async functions, not just async arrow functions.

```js
async function foo () {}
```
```js
var foo = async function () {}
```
```js
var obj = {
  foo: async function () {}
}
```
```js
var obj = {
  async foo () {}
}
```

**Comment by loganfsmyth** [comment]
Sorry! There is a regression in ESLint 2.10.0: https://github.com/eslint/eslint/issues/6158 so for not stick to 2.9.x and wait for the fix.

**Comment by hzoo** [comment]
ESLint 2.10.1 is released - if someone wants to verify, we can close.

**Comment by not-an-aardvark** [comment]
Can confirm this is fixed with ESLint 2.10.1.

**Comment by hzoo** [comment]
Awesome.
